action.yml

```diff
@@ -84,14 +84,64 @@ branding:
   color: yellow
   icon: upload-cloud
 runs:
-  using: docker
-  image: Dockerfile
-  args:
-  - ${{ inputs.user }}
-  - ${{ inputs.password }}
-  - ${{ inputs.repository-url }}
-  - ${{ inputs.packages-dir }}
-  - ${{ inputs.verify-metadata }}
-  - ${{ inputs.skip-existing }}
-  - ${{ inputs.verbose }}
-  - ${{ inputs.print-hash }}
+  using: composite
+  steps:
+  - name: Reset path if needed
+    run: |
+      # Reset path if needed
+      # https://github.com/pypa/gh-action-pypi-publish/issues/112
+      if [[ $PATH != *"/usr/bin"* ]]; then
+        echo "\$PATH=$PATH. Resetting \$PATH for GitHub Actions."
+        PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
+        echo "PATH=$PATH" >>"$GITHUB_ENV"
+        echo "$PATH" >>"$GITHUB_PATH"
+        echo "\$PATH reset. \$PATH=$PATH"
+      fi
+    shell: bash
+  - name: Set repo and ref from which to run Docker container action
+    id: set-repo-and-ref
+    run: |
+      # Set repo and ref from which to run Docker container action
+      # to handle cases in which `github.action_` context is not set
+      # https://github.com/actions/runner/issues/2473
+      REF=${{ env.ACTION_REF || env.PR_REF || github.ref_name }}
+      REPO=${{ env.ACTION_REPO || env.PR_REPO || github.repository }}
+      REPO_OWNER_ID=${{ env.PR_REPO_OWNER_ID || github.repository_owner_id }}
+      echo "ref=$REF" >>"$GITHUB_OUTPUT"
+      echo "repo=$REPO" >>"$GITHUB_OUTPUT"
+      echo "repo-owner-id=$REPO_OWNER_ID" >>"$GITHUB_OUTPUT"
+    shell: bash
+    env:
+      ACTION_REF: ${{ github.action_ref }}
+      ACTION_REPO: ${{ github.action_repository }}
+      PR_REF: ${{ github.event.pull_request.head.ref }}
+      PR_REPO: ${{ github.event.pull_request.head.repo.full_name }}
+      PR_REPO_OWNER_ID: ${{ github.event.pull_request.base.repo.owner.id }}
+  - name: Check out action repo
+    uses: actions/checkout@v4
+    with:
+      path: action-repo
+      ref: ${{ steps.set-repo-and-ref.outputs.ref }}
+      repository: ${{ steps.set-repo-and-ref.outputs.repo }}
+  - name: Create Docker container action
+    run: |
+      # Create Docker container action
+      python create-docker-action.py
```

Comment:
No need to copy-paste the step description as a code comment, it doesn't add any value and would diverge over time.
```suggestion
    run: python create-docker-action.py
```

Reply:
I'm fine with removing the actions step comments, but I don't agree that "it doesn't add any value." The comments are helpful because [GitHub Actions doesn't log steps clearly for composite actions](https://github.com/orgs/community/discussions/21276). It groups logs from all the steps together and doesn't log step names. The logs become difficult to read. The comments help because they are like step names.

To test this, I published to the GitHub Marketplace from my fork. Here's how the logs look when running the Marketplace action.

<img src="https://github.com/pypa/gh-action-pypi-publish/assets/26674818/ed3aa508-441a-4636-857f-013f523434a4" alt="Screenshot of GitHub Actions logs from composite action" width="75%" />

Reply:
Oh, I never thought of it like that! In some other actions, I've been trying to generate the groups manually. Alright, don't change it for now. I'm marking this thread as resolved.

P.S. You *don't have to* publish to Marketplace to test the action. You can reference any repository+committish (branch/tag/commit) regardless of whether it's on Marketplace or not.